Snail/Observable.swift

```diff
@@ -89,7 +89,7 @@ public class Observable<T>: ObservableType {
     public func map<U>(_ transform: @escaping (T) -> U) -> Observable<U> {
         let transformed = Observable<U>()
 
-        subscribe(
+        _ = subscribe(
```

Comment:
I think we should add this to the subscribers array right? This way we remove this subscription when `deinit`

Reply:
`.subscribe()` adds the subscription to the subscribers array.